src/sas/sascalc/poresize/SizeDistribution.py

```diff
@@ -79,22 +83,37 @@ def background_fit(data, power=None, qmin=None, qmax=None, type="fixed"):
     fx[idx] = data.y[idx]
 
     ##Get values of scale and if required power
-    if power is not None and power != 0:
-        # Linearize the data for a power law fit (log, log)
-        linearized_data = Data1D(np.log(data.x[idx]), np.log(fx[idx]), dy=(1/np.log(10))*sigma[idx]/fx[idx])
+    if power is not None:
+        # Linearize the data for a linear or power law fit
+        
+        linearized_data = Data1D(np.log(data.x[idx]), np.log(fx[idx]), dy=sigma[idx]/fx[idx])
+        fit_func = lambda x,b:line_func(x, power, b)
+        init_guess = (linearized_data.y[0])
+
     else:
-        linearized_data = Data1D(data.x[idx], fx[idx], dy=None)
+        # Fit both the power and scale 
+        linearized_data = Data1D(np.log(data.x[idx]), np.log(fx[idx]), dy=sigma[idx]/fx[idx])
```

Comment:
This is duplicated inside both cases of the if/else block and can be moved out to before the if statement.